tools/testtool/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"dagger.io/dagger"
+)
+
+func main() {
+	sandbox := flag.Bool("sandbox", false, "Run the provided command in a sandboxed environment instead of your native environment")
+	flag.Parse()
+	if flag.NArg() < 2 {
+		log.Fatalln("too few arguments")
+	}
+
+	ctx := context.Background()
+	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stdout))
+	if err != nil {
+		log.Fatalf("unable to connect to client: %v\n", err)
+	}
+	defer client.Close()
+
+	var kernel *dagger.File
+	var kpath, qemuCmd string
+
+	switch envWithDefault("UROOT_TESTARCH", runtime.GOARCH) {
+	case "amd64":
+		kpath = "/bzImage"
+		qemuCmd = "qemu-system-x86_64 -L %s -m 1G"
+		kernel = client.Container().From("ghcr.io/hugelgupf/vmtest/kernel-amd64:main").File(kpath)
+	case "arm":
+		kpath = "/zImage"
+		qemuCmd = "qemu-system-aarch64 -machine virt -cpu max -m 1G -L %s"
+		kernel = client.Container().From("ghcr.io/hugelgupf/vmtest/kernel-arm:main").File(kpath)
+	case "arm64":
+		kpath = "/Image"
```

Comment:
here's what I'd do. The following makes the default values of flags be based on VMTEST_GOARCH / runtime.GOARCH, but overridable by flags so users can specify everything just in case.

```go
type TestEnvConfig struct {
  KernelContainer string
  KernelPath string
  QEMUContainer string
  QEMUCmd string
  BIOSPath string
}

func (tc *TestEnvConfig) RegisterFlags(f *flag.FlagSet) {
  // Note the default value is whatever is in tc already.
  f.StringVar(&tc.KernelContainer, "kernel-container", tc.KernelContainer, "Container to use for kernel files")
  ...
}

var configs = map[string]TestEnvConfig{
  "amd64": TestEnvConfig{
    KernelContainer: "ghcr.io/hugelgupf/vmtest/kernel-amd64:main",
    KernelPath: "/bzImage",
    QEMUContainer: "ghcr.io/hugelgupf/vmtest/qemu:main",
    QEMUCmd: "qemu-system-x86_64 -L %s -m 1G",
    QEMUPath: "/zqemu/bin/qemu-system-x86_64",
    BIOSPath: "/zqemu/pc-bios",
  },
  ...
}

func defaultConfig() TestEnvConfig {
  arch := os.GetEnv("UROOT_TESTARCH")
  if c, ok := configs[arch]; ok {
    return c
  }
  if c, ok := configs[runtime.GOARCH]; ok {
    return c
  }
  // On other architectures, user has to provide all values via flags.
  return TestEnvConfig{}
}

func main() {
  config := defaultConfig()
  config.RegisterFlags(flag.CommandLine)
  flag.Parse()

  ...
}
```